### PR TITLE
Working around #1965 in UIImageSetLayerContents.

### LINKED ISSUE #1962: UIKit tests not reliable
UIButtonTests::UIButton_CheckForLeaks occasionally fails with:

FAULTING_SOURCE_LINE:  y:\arcadia\winobjc\frameworks\uikit\starboardxaml\layerproxy.cpp

FAULTING_SOURCE_FILE:  y:\arcadia\winobjc\frameworks\uikit\starboardxaml\layerproxy.cpp

FAULTING_SOURCE_LINE_NUMBER:  137

FAULTING_SOURCE_CODE:  
   133: }
   134: 
   135: void LayerProxy::_SetContentsCenter(float x, float y, float width, float height) {
   136:     FrameworkElement^ xamlLayer = _FrameworkElementFromInspectable(_xamlElement);
>  137:     CoreAnimation::LayerCoordinator::SetContentCenter(xamlLayer, Rect(x, y, width, height));
   138: }
   139: 
   140: // TODO: Find a way to remove this
   141: void LayerProxy::SetTopMost() {
   142:     FrameworkElement^ xamlLayer = _FrameworkElementFromInspectable(_xamlElement);

### PR DESCRIPTION
**Pull request by jaredhms**
Fixes #1962.

### CODE REVIEW

File: Frameworks/UIKit/UIImage.mm [hunk 1]
```diff
@@ -69,7 +69,9 @@ void UIImageSetLayerContents(CALayer* layer, UIImage* image) {
     if ([layer contentsOrientation] != [image imageOrientation]) {
         [layer setContentsOrientation:[image imageOrientation]];
     }
-    CGRect stretch = [image _imageStretch];
```
Comment (yiyang-msft) [inline Frameworks/UIKit/UIImage.mm]: >stretch [](start = 11, length = 7)

init stretch to CGRectZero should avoid the issue as well. i have seen some code pattern like that in our code base.

**Comment by yiyang-msft** [comment]
:shipit:

File: Frameworks/UIKit/UIImage.mm [hunk 2]
```diff
@@ -69,7 +69,9 @@ void UIImageSetLayerContents(CALayer* layer, UIImage* image) {
...
+
+    // Working around #1965; only call '_imageStretch' if 'image' is non-nil
+    CGRect stretch = (image ? [image _imageStretch] : CGRectZero);
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UIImage.mm]: >( [](start = 21, length = 1)

You don't need the outer parenthesis.

**Comment by oliversa-msft** [comment]
:shipit:

File: Frameworks/UIKit/UIImage.mm (hunk 2 above)
Comment (jaredhms) [inline Frameworks/UIKit/UIImage.mm]: agreed; they are there for clarity.

**Pull request merged**